scripts/globals/job_utils/rune_fencer.lua

```diff
@@ -50,8 +50,163 @@ local function enforceRuneCounts(target)
+            type == xi.effect.BLINDNESS or type == xi.effect.SILENCE or type == xi.effect.MUTE or type == xi.effect.CURSE_I or
+            type == xi.effect.CURSE_II or type == xi.effect.DOOM or type == xi.effect.VIRUS or
+            type == xi.effect.PLAGUE or type == xi.effect.PETRIFICATION
+        then
+            debuffs[debuffCount+1] = type
+        debuffCount = debuffCount + 1
+        end
+
+        if type == xi.effect.TENEBRAE then -- runes that also restore MP
+            tenebraeRuneCount = tenebraeRuneCount + 1
+        end
+    end
+
+    if tenebraeRuneCount > 0 then -- only restore MP if there's one or more tenebrae rune active
+        local MPHealAmount = math.floor(divineMagicSkillLevel / 10 * (100 + target:getJobPointLevel(xi.jp.VIVACIOUS_PULSE_EFFECT)) / 100) * (tenebraeRuneCount + 1)
+        target:addMP(MPHealAmount) -- augment bonusPct does not apply here according to testing.
+    end
+
+    if debuffCount > 0 and (target:getEquipID(xi.slot.HEAD) == 26782 or target:getEquipID(xi.slot.HEAD) == 26783) then -- add random removal of Poison, Paralyze, Blind, Silence, Mute, Curse, Bane, Doom, Virus, Plague, Petrification via AF3 head (source: https://www.bg-wiki.com/ffxi/Erilaz_Galea)
+        target:delStatusEffect(debuffs[math.random(debuffCount)])
+    end
+
+    HPHealAmount = HPHealAmount * bonusPct
+    if (target:getHP() + HPHealAmount > target:getMaxHP()) then
+        HPHealAmount = target:getMaxHP() - target:getHP() -- don't go over cap
+    end
+
+    return HPHealAmount
+end
 
 xi.job_utils.rune_fencer.useRuneEnchantment = function(player, target, ability, effect)
     enforceRuneCounts(target)
     applyRuneEnhancement(effect, target)
 end
+
+xi.job_utils.rune_fencer.useSwordplay = function(player, target, ability)
+    local power = 0 --Swordplay starts at +0 bonus without swaps
+
+    -- see https://www.bg-wiki.com/ffxi/Sleight_of_Sword for levels
+    local meritBonus =  player:getMerit(xi.merit.MERIT_SLEIGHT_OF_SWORD)
+    local augBonus = 0 -- augBonus = 2 per level of merit
+
+    -- gear bonuses from https://www.bg-wiki.com/ffxi/Swordplay
+    if player:getMainJob() == xi.job.RUN and target:getMainLvl() == 99 then -- don't bother with gear boost checks until 99 and main RUN
+        local tickPower = 3 -- Tick power appears to be 3/tick, not 6/tick if RUN main and 3/tick if RUN sub; source : https://www.ffxiah.com/forum/topic/37086/endeavoring-to-awaken-a-guide-to-rune-fencer/180/#3615377
+        local handsSlotID = target:getEquipID(xi.slot.HANDS)
+
+        -- add starting tick bonuses if appropriate gear is equipped
+        if handsSlotID      == 27018 then
+            power = 3 * tickPower
+        elseif handsSlotID  == 27019 then
+            power = 5 * tickPower
+        elseif handsSlotID  == 23218 then
+            power = 7 * tickPower
+        elseif handsSlotID  == 23553 then
+            power = 9 * tickPower
+        end
+    end
```

Comment:
These will need to be extracted out to an item mod, if you let us know what you need here we can push those changes straight into main so you don't get any conflicts while this branch is ongoing

Reply:
![image](https://user-images.githubusercontent.com/60417494/155935541-35389d37-06dd-4447-a8f0-0f5b95133f61.png)
"Swordplay" item mod makes sense. I don't think anything else but this particular job specific gear boosts this currently though. These item IDs are all from Futhark Mitons/Futhark Mitons +1/Futhark Mitons +2/Futhark Mitons +3. We'd also need to get the extdata for Sleight of Sword if we replace that with an item mod, since we are no longer "guaranteed" to have the item equipped that will always have said augment. That, or the ID check stays but only for the Sleight of Sword augment bonus.

I have Futhark Mitons +3 in retail so i'll be able to dump the packet/data for the augment if I get instructions on how to do so. I assume NQ and all HQs all share the same augment...

Reply:
We -always- want to avoid checking item IDs when possible. And there's always the Chance SE adds something new that would use the mod.

> We'd also need to get the extdata for Sleight of Sword if we replace that with an item mod,

That's not that bad. The part that may be a problem is if its in the segment of augment IDs we handle incorrectly atm (some don't display right, but its usually the type that use a the other items typing with the sockets or slotting). That said, I would bet on its ID being nearby the other job-ability enchanting augments already in our table. Once we have the ID its just a matter of filling in the modifier info for it in the aforementioned table.

Reply:
I'll see if I can find a cap or even dig through the values for it later.

Reply:
Turns out we already have the ID
```sql
INSERT INTO `augments` VALUES (1457,0,0,0,0,0); -- Enhances "Sleight of Sword" effect
```
just needs modifier hooked up where those zeroes are, in `augments.sql`

Reply:
yup, client agrees. I added that aug to an !additem call.
![image](https://user-images.githubusercontent.com/60417494/155938914-b02d2d63-d6c4-4e59-8d2b-1dd9b9039813.png)

I will need mods for the following:

- Enhances "Battuta" effect
- Enhances "Elemental Sforzo" effect 
- Enhances "Sleight of Sword" effect
- Enhances "Inspiration" effect 
- Enhances "Elemental Sforzo" effect 
- Swordplay
- Liement
- "Enhances "Valiance" and "Vallation" effects" (they are literally the same buff, but one is AoE. Only one mod is needed.)
- Pflug
- "Liement" effect extends to an area
- "Vivacious Pulse" potency (looks like the augment is 1310 according to windower resources, but I can't get Morgelai to display it. Probably some nonsense with the Path A/B/C extdata that I know very little about)

Reply:
> "Vivacious Pulse" potency (looks like the augment is 1310 according to windower resources, but I can't get Morgelai to display it. Probably some nonsense with the Path A/B/C extdata that I know very little about)

Windower's resource extractor pulls the dat file index, while our table uses the order the packet does - the 2 lists don't agree and we currently lack a way to match them up. Take a look in the augments.sql file and use the `!additem` GM command to try nearby IDs starting from where "Slight of Sword" was - its probably nearby.